### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by manjunathc23**
Upgrade dagger, play services, support libs and typo in the `dependencies.gradle`

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -50,7 +52,7 @@ ext.versions = [
         robolectricVersion                : '3.0',
         assertJVersion                    : '1.7.1',
         equalsVerifierVersion             : '1.7.5',
-        mockitoVersion                    : '1.10.19',
+        mockitoVersion                    : 'mockito-core:1.10.19',
```
Comment (manijshrestha) [inline dependencies.gradle]: any reason to move `mockito-core` part here? Let's move it back in the dependencies block?

File: dependencies.gradle [hunk 2]
```diff
@@ -62,6 +64,7 @@ ext.versions = [
 ]
 
 ext.gradlePlugins = [
+        aptPlugin         : "com.neenbedankt.gradle.plugins:android-apt:1.8",
```
Comment (manijshrestha) [inline dependencies.gradle]: 🤔 , this was removed in favor of build in `annotationProcessor`. any reason to bringing it back?

File: dependencies.gradle [hunk 3]
```diff
@@ -88,7 +91,10 @@ ext.libraries = [
         playServicesMaps           : "com.google.android.gms:play-services-maps:$versions.playServicesVersion",
         playServicesPlaces         : "com.google.android.gms:play-services-places:$versions.playServicesVersion",
         playServicesLocation       : "com.google.android.gms:play-services-location:$versions.playServicesVersion",
-        appIndexing                : "com.google.android.gms:play-services-appindexing:$versions.playServicesVersion",
+        //endregion
+
+        //region firebase
+        appIndexing                : "com.google.firebase:firebase-appindexing:$versions.playServicesVersion",
```
Comment (manijshrestha) [inline dependencies.gradle:96]: 👍

File: dependencies.gradle [hunk 4]
```diff
@@ -2,10 +2,10 @@ ext.versions = [
         compileSdk                        : 24,
         minSdk                            : 16,
         targetSdk                         : 23,
```
Comment (manijshrestha) [inline dependencies.gradle:4]: Lets bump this and compileSdk to 25?

File: dependencies.gradle [hunk 5]
```diff
@@ -2,10 +2,10 @@ ext.versions = [
...
-        buildTools                        : "24.0.3",
+        buildTools                        : "25.0.1",
```
Comment (manijshrestha) [inline dependencies.gradle]: lets even go up to `25.0.2`?

File: dependencies.gradle [hunk 6]
```diff
@@ -22,12 +22,14 @@ ext.versions = [
...
-        daggerVersion                     : "2.5",
-        retrofitVersion                   : "2.0.2",
+        daggerVersion                     : "2.8",
+        retrofitVersion                   : "2.1.0",
         rxJavaVersion                     : "1.2.1",
         rxAndroidVersion                  : "1.2.1",
 
         gsonVersion                       : "2.7",
+        cookiesJarVersion                 : "v1.0.0",
+        networkSamplerVersion             : "connectionclass:1.0.1",
```
Comment (manijshrestha) [inline dependencies.gradle:32]: lets move the artifactId to dependencies block?

File: dependencies.gradle [hunk 7]
```diff
@@ -155,18 +164,19 @@ ext.libraries = [
         //endregion
 
         //region TESTING DEPENDENCIES
+        orgDotJson                 : "org.json:json:20140107",
```
Comment (manijshrestha) [inline dependencies.gradle]: 🤔 would we need this? If we do, shall we use the latest version? `20160810`?

**Review by manijshrestha — CHANGES_REQUESTED** [review]
Looks good!

File: dependencies.gradle (hunk 4 above)
Comment (manjunathc23) [reply dependencies.gradle:4]: Sure 👍

File: dependencies.gradle (hunk 5 above)
Comment (manjunathc23) [reply dependencies.gradle]: Yep 👍

File: dependencies.gradle (hunk 6 above)
Comment (manjunathc23) [reply dependencies.gradle:32]: For some reason it wont compile if I do so. Can you check at your end too?

File: dependencies.gradle (hunk 1 above)
Comment (manjunathc23) [reply dependencies.gradle]: compilation reason. Can you check from your end too?

File: dependencies.gradle (hunk 2 above)
Comment (manjunathc23) [reply dependencies.gradle]: May be butterknife needs it. Will confirm you.

File: dependencies.gradle (hunk 7 above)
Comment (manjunathc23) [reply dependencies.gradle]: We can remove this. Not currently in use.

**Comment by manjunathc23** [comment]
@manijshrestha  ^ take a look now. New commits are up.

**Review by manijshrestha — APPROVED** [review]
💯

**Comment by manijshrestha** [comment]
Looks good. Lets move the artifact-id down to the dependencies block and it all looks good.

**Comment by manjunathc23** [comment]
Closing this PR as i'll raise the new one. Thanks @manijshrestha

**Pull request closed without merging**